Sprint_6

garini491 opened the pull request.

garini491 committed:
Написаны тесты для классов Lion, Feline, Cat. Покрытие согласно jacoco.exec 100%

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -16,7 +16,10 @@ public Lion(String sex) throws Exception {
 
-    Feline feline = new Feline();
+    private Feline feline;
+    public Lion(Feline feline) {
```
изоляцию Lion от Feline нужно было делать через инъекцию зависимостей. То есть в класс Lion должен попадать уже созданный экземпляр класса Feline.

Тебе нужно реализовать один конструктор public Lion(String sex, Feline feline).

Это нужно исправить.

alesyaisachenko commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,61 @@
+        this.sex = sex;
+        this.hasMane = hasMane;
+    }
+    @Parameterized.Parameters
```
Параметризованные тесты лучше выносить в отдельный класс.
Сейчас у тебя не параметризованные тесты запускаются несколько раз подряд -- это трата ресурсов.

Для параметризованного теста создай, пожалуйста, отдельный тестовый класс.

alesyaisachenko commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,33 @@
+    }
+    @Test
+    public void getFoodReturnsPredator() throws Exception {
+        Mockito.when(predator.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
```
сейчас на этом месте тест падает -- все тесты должны проходить.

garini491 committed:
Вынес непараметризованные тесты в отдельный класс
Добавил имя к аннотации
Добавил feline в конструктор Lion
Тесты Cat теперь не падают из-за mockito
Добавил отчет jacoco как описано в задании
